pkg/cloud/azureprovider.go

```diff
@@ -740,10 +839,55 @@ func (az *Azure) AllNodePricing() (interface{}, error) {
 func (az *Azure) NodePricing(key Key) (*Node, error) {
 	az.DownloadPricingDataLock.RLock()
 	defer az.DownloadPricingDataLock.RUnlock()
+
 	azKey, ok := key.(*azureKey)
 	if !ok {
 		return nil, fmt.Errorf("azure: NodePricing: key is of type %T", key)
 	}
+	config, _ := az.GetConfig()
+	if config.SpotLabel != "" && config.SpotLabelValue != "" {
+		labelsMapStr := fmt.Sprintf("%s", azKey)
+		spotLabelMapStr := fmt.Sprintf(" %s:%s ", config.SpotLabel, config.SpotLabelValue)
+		if strings.Contains(labelsMapStr, spotLabelMapStr) {
+			features := strings.Split(azKey.Features(), ",")
+			spotFeatures := strings.ReplaceAll(azKey.Features(), ",ondemand", ",spot")
+			if n, ok := az.Pricing[spotFeatures]; ok {
+				klog.V(4).Infof("Returning pricing for node %s: %+v from key %s", azKey, n, spotFeatures)
+				if azKey.isValidGPUNode() {
+					n.Node.GPU = "1" // TODO: support multiple GPUs
+				}
+				return n.Node, nil
+			}
+			klog.V(1).Infof("[Info] found spot instance, trying to get retail price for %s: %s, ", spotFeatures, azKey)
+
+			spotCost, err := getRetailPrice(features[0], features[1], config.CurrencyCode, true)
```

Comment:
Is the order (mapping of index -> feature) of `features` always the same? Is there any danger of `len(features) < 2`?